probes/blocksDeleteOnBranches/def.yml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: blocksDeleteOnBranches
+short: Check that the project blocks non-admins from deleting branches.
+motivation: >
+  Allowing non-admins to delete project branches could prevent users from consuming the project.
```

Comment:
my understanding is that branch deletion + creation is equivalent to a force push. So I thought this reason was identical to blocks Force Push motivation.

Reply:
https://github.com/ossf/scorecard/pull/3691/commits/c6d146ae888b4396c8af1bf3ea6f2cca25b88d12